### ISSUE TITLE
Cannot find solver executable file (Ubuntu 20.04)

### ISSUE THREAD

**Issue by AboodaA** [energyRt/energyRt#41]
Hello. The below applies to a use of the development version as of last week, running on Ubuntu 20.04. In short, I simply cannot get energyRt to detect the  executable file of the solver. Here is some explanation of what I did. 

If I set the solver to be Julia (let's say) and point towards the location of the JuMP package like this: 
set_julia_path("/.../MYNAME/.julia/packages/JuMP/")
I also then set the default solver to be Julia and then pretty much ran the script provided in the RMD file (available on this repo). The error I received was: 

sh: 1: /.../.julia/packages/JuMP//julia: not found
Error in .call_solver(arg, scen) : Solution error code 127

My understanding is that this means that it cannot find the correct file to execute. I have ensured though that this is the correct location of the Julia CBC file. I have also tried pointing to the julia-1.8.1/bin/julia location and a few other things. 

I have also tried similar steps getting other solves to work. In general the order of operations seems to work because the folders are created in the working directory where I expected them to be. I have set everything up as explained in the vignette document and have set up various solvers, but energyRt simply cannot find it. 

My best guess is that this is something broadly to do with R and/or energyRt, perhaps energyRt in Ubuntu? I have tested out pyomo and Julia separately to ensure that they work. 

I'd be very grateful for your advice on how to proceed with this; I'd love to make use of energyRt.

**Comment by olugovoy** [comment]
Hello and thanks for reporting!
The set_julia_path() function must point to julia executable, not julia packages. Packages are called from inside julia (or python), but we need to point energyRt to the right executable.
I would expect `set_julia_path("julia-1.8.1/bin")`  to work if this directory has julia executable. 
If this doesn't work, please send the reported error again.

**Comment by olugovoy** [comment]
Please also check if all Julia packages have been installed:
<https://ideea-model.github.io/IDEEA.dev/articles/install.html>
(the IDEEA package from the link is not public yet, but will be in a week or two)

**Comment by olugovoy** [comment]
Another way to check that Julia (Python, etc.) is working correctly is to run the generated model from the command line:
`mod <- newModel(...) `
`mod <- interpolate(mod)`
`mod <- write_script(mod, tmp.dir = "DIRECTORY FOR THE MODEL SCRIPT", solver = solver_options$julia_highs)`
Then in terminal navigate to the `tmp.dir`, check if the model files are there and run:
`julia energyRt.jl`
Once solved, continue in R:
`mod <- read_solution(mod)`

**Comment by olugovoy** [comment]
I don't have an access to Linux right now, but it looks like the "julia-1.8.1/bin/julia" is not a full path. Is it responding from terminal if you run this line? If julia responds then "julia-1.8.1/bin/" will be the path. Also, you can add it to PATH, here is more info:
[How to Install Julia Programming Language on Ubuntu 22.04](https://www.digitalocean.com/community/tutorials/how-to-install-julia-programming-language-on-ubuntu-22-04).

**Comment by AboodaA** [comment]
Thank you for your suggestions and, especially, for taking time out of your weekend for them. I will work through these and get back to you.

**Comment by AboodaA** [comment]
Hello again.  Thank you so much for all of your advice. I did manage to get it to run; what worked in the end was to move to the JULIA executable to the /usr/bin folder; even when I had JULIA in my PATH environment (i.e., echo $PATH returned a folder where a Julia executable was), this simply did not work. I am not sure if this is just my set up or something to do with Ubuntu 20.04. I did notice that there were some spare front slashes here and there; I will look to ensure if it is indeed coming from energy Rt or if it's due to something foolish which I'm doing at my end. If appropriate, I'll put that in a separate issue. 

Another thing to keep in mind that the JULIA working is based on installing the RData package, which I did not anticipate. I may have missed something in the introductory materials, but I could not find it there.
